app/tasks/raster_tile_cache_assets/raster_tile_cache_assets.py

```diff
@@ -21,6 +25,54 @@
 from app.utils.path import get_asset_uri
 
 
+def generate_stats(asset_id) -> RasterStats:
+    raise NotImplementedError()
+
+
+def convert_float_to_int(
+    asset_id: UUID,
+    stats: Optional[Dict[str, Any]],
+    source_asset_co: RasterTileSetSourceCreationOptions,
+) -> Tuple[RasterTileSetSourceCreationOptions, str]:
+    if stats is None:
+        stats = generate_stats(asset_id)
+    else:
+        stats = RasterStats(**stats)
+    assert len(stats.bands) == 1
+    stats_min = stats.bands[0].min
+    stats_max = stats.bands[0].max
+    value_range = math.fabs(stats_max - stats_min)
+
+    # Shift by 1 (and add 1 later) so any values of zero don't get counted as no_data
+    uint16_max = np.iinfo(np.uint16).max - 1
+    # Expand or squeeze to fit into a uint16
+    mult_factor = int(math.floor(uint16_max / value_range)) if value_range else 1
+
+    old_no_data = source_asset_co.no_data
+    if old_no_data is None:
+        old_no_data = "None"
+    elif old_no_data == str(np.nan):
+        old_no_data = "np.nan"
+    else:
+        old_no_data = float(old_no_data)
+
+    calc_str = (
+        f"(A != {old_no_data}).astype(np.uint8) * "
+        f"(1 + (A - {stats_min}) * {mult_factor}).astype(np.uint16)"
+    )
+
+    source_asset_co.data_type = DataType.uint16
+    source_asset_co.no_data = 0
```

Comment:
I think you might be overthinking this?

PixETL assigns RGBA(0,0,0,0) to a the no data value when applying a symbology
https://github.com/wri/gfw_pixetl/blob/master/gfw_pixetl/tiles/tile.py#L254-L257

np.nan * x always returns np.nan
```suggestion
    calc_str = (
        f"(1 + (A - {stats_min}) * {mult_factor}).astype(np.uint16)"
    )

    source_asset_co.data_type = DataType.uint16
    source_asset_co.no_data = int(1 + (source_asset_co.no_data - {stats_min}) * {mult_factor})
```